src/components/ListItemEditForm.js

```diff
@@ -1,74 +1,95 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import {
-  Button,
-  ControlLabel,
-  Form,
-  FormControl,
-  FormGroup,
-  OverlayTrigger,
-  Tooltip
-} from 'react-bootstrap';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
-const styles = {
-  formControl: {
-    margin: '0 5px'
-  }
-};
+import { connect } from 'react-redux';
+import TimespanInlineForm from './TimespanInlineForm';
+import HeadingInlineForm from './HeadingInlineForm';
+import { buildSMUI } from '../actions/sm-data';
+import { cloneDeep } from 'lodash';
+import StructuralMetadataUtils from '../services/StructuralMetadataUtils';
 
-const tooltip = tip => <Tooltip id="tooltip">{tip}</Tooltip>;
+const structuralMetadataUtils = new StructuralMetadataUtils();
 
 class ListItemEditForm extends Component {
+  constructor(props) {
+    super(props);
+    this.type = this.props.item.type;
+    this.id = this.props.item.id;
+  }
+
   static propTypes = {
     handleEditFormCancel: PropTypes.func,
     item: PropTypes.object.isRequired
   };
 
+  addUpdatedValues(item, payload) {
+    if (item.type === 'div') {
+      item.label = payload.headingTitle;
+    } else if (item.type === 'span') {
+      item.label = payload.timespanTitle;
+      item.begin = payload.beginTime;
+      item.end = payload.endTime;
+    }
+
+    return item;
+  }
+
   handleCancelClick = e => {
     this.props.handleEditFormCancel();
   };
 
+  handleSaveClick = (id, payload) => {
+    // Clone smData
+    let clonedItems = cloneDeep(this.props.smData);
+
+    // Get the original item
+    /* eslint-disable */
+    let item = structuralMetadataUtils.findItem(id, clonedItems);
+    /* eslint-enable */
+
+    // Update item values
+    item = this.addUpdatedValues(item, payload);
+
+    // Send updated smData back to redux
+    this.props.buildSMUI(clonedItems);
```

Comment:
@adamjarling How does the updated item get added back to `clonedItems` array?

Reply:
So for this example, the `item` variable is a reference to the list item object within the data structure.  I might not even need to re-assign the value back to `item` come to think of it.  It's updating the nested object within the data structure, then the updated object is included in `clonedItems`.

Reply:
And if you noticed in this file, I think I pulled in the `cloneDeep()` lodash function.  This is how we should be 'cloning' that `smData` object... by doing deep clones.  The way I was doing it before, ie. `let clonedItems = [...smData]` doesn't create a deep clone and chances are it was mutating data it wasn't intended to.   I'm going to write up an issue for this and fix what I started:)

Reply:
Thanks @adamjarling !